src/main/java/org/veupathdb/service/eda/ds/plugin/CategoricalCountsPlugin.java

```diff
@@ -0,0 +1,140 @@
+package org.veupathdb.service.eda.ds.plugin;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.gusdb.fgputil.cache.ManagedMap;
+import org.gusdb.fgputil.DelimitedDataParser;
+import org.gusdb.fgputil.EncryptionUtil;
+import org.gusdb.fgputil.json.JsonUtil;
+import org.gusdb.fgputil.ListBuilder;
+import org.gusdb.fgputil.validation.ValidationException;
+import org.json.JSONObject;
+import org.veupathdb.service.eda.common.client.spec.StreamSpec;
+import org.veupathdb.service.eda.common.plugin.constraint.ConstraintSpec;
+import org.veupathdb.service.eda.common.plugin.constraint.DataElementSet;
+import org.veupathdb.service.eda.common.plugin.util.PluginUtil;
+import org.veupathdb.service.eda.ds.Resources;
+import org.veupathdb.service.eda.ds.plugin.AbstractEmptyComputePlugin;
+import org.veupathdb.service.eda.generated.model.APIVariableType;
+import org.veupathdb.service.eda.generated.model.APIVariableDataShape;
+import org.veupathdb.service.eda.generated.model.CategoricalCountsSpec;
+import org.veupathdb.service.eda.generated.model.CategoricalCountsPostRequest;
+import org.veupathdb.service.eda.generated.model.CategoricalCountsPostResponse;
+import org.veupathdb.service.eda.generated.model.VariableSpec;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+import static org.gusdb.fgputil.FormatUtil.TAB;
+
+public class CategoricalCountsPlugin extends AbstractEmptyComputePlugin<CategoricalCountsPostRequest, CategoricalCountsSpec> {
+  
+  private static final Logger LOG = LogManager.getLogger(CategoricalCountsPlugin.class);
+
+  private static final ManagedMap<String,CategoricalCountsPostResponse> RESULT_CACHE =
+      new ManagedMap<>(5000, 2000);
+
+  private String _cacheKey;
+  private CategoricalCountsPostResponse _cachedResponse;
+
+  @Override
+  protected Class<CategoricalCountsPostRequest> getVisualizationRequestClass() {
+    return CategoricalCountsPostRequest.class;
+  }
+
+  @Override
+  protected Class<CategoricalCountsSpec> getVisualizationSpecClass() {
+    return CategoricalCountsSpec.class;
+  }
+
+  @Override
+  public ConstraintSpec getConstraintSpec() {
```

Comment:
i thought about this some when i was drafting this.. and im not sure we do need to send these to the client? i mean they arent constraints in the sense weve meant up to this point since this isnt a viz where were trying to constrain some user selection.